2.1-Segwit-Version-1.ipynb

```diff
@@ -0,0 +1,564 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Spending a Segwit v1 Output with the Key Path\n",
+    "\n",
+    "Unlike Segwit v0 outputs, v1 outputs are spendable along two paths. In this chapter we will focus on spending the key path.\n",
+    "\n",
+    "![test](images/segwit_version1_0.jpg)\n",
+    "\n",
+    " * **Key path:**\n",
+    "     * Spendable along the key path.\n",
+    "     * Spending witness: **`[sig]`**\n",
+    " * **Script path:**\n",
+    "     * If public key is tweaked with a valid taproot.\n",
+    "     * (See chapters 2.1 and 2.2)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 2.1.1 Example: Constructing a Segwit V1 Output\n",
+    "\n",
+    "In this example, we construct segwit v1 output for spending along the key path.\n",
+    "* Generate a key pair.\n",
+    "* Encode public key (Segwit V1 program).\n",
+    "* Derive Bech32 address.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Key Pair generation\n",
+    "sec = ECKey()\n",
+    "sec.generate()\n",
+    "pubkey = sec.get_pubkey()\n",
+    "\n",
+    "# Create Segwit program\n",
+    "pubkey_data = pubkey.get_bytes()\n",
+    "program = bytes([pubkey_data[0] & 1]) + pubkey_data[1:]\n",
+    "\n",
+    "# Create (regtest) Bech32 address\n",
+    "version = 0x01\n",
+    "address = segwit_addr.encode(\"bcrt\", version, program)\n",
+    "print(address)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Part 2. Sending funds from the Bitcoin Core wallet\n",
+    "\n",
+    "We now send funds to a Segwit v1 output, in order to demonstrate how to spend a taproot output along the key path.\n",
+    "\n",
+    "* Start up the Regtest node and wallet.\n",
+    "* Generate Regtest coins for the wallet.\n",
+    "* Send a wallet transaction to the Segwit v1 output."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Example 2.1.2: Startup TestWrapper to initialize a Regtest node and wallet\n",
+    "* Run setup only once, or after a clean shutdown."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test = util.TestWrapper()\n",
+    "test.setup(num_nodes=1)\n",
+    "\n",
+    "version = test.nodes[0].getnetworkinfo()['subversion']\n",
+    "print(\"Client version is {}\".format(version))\n",
+    "assert version == '/OptechTaproot:0.1.1(testnode0)/'"
```

Comment:
Needs updating to 0.1.2

Reply:
I just removed this entirely, so we don't need to keep updating these